### ISSUE TITLE
$0 costs for azurerm_kubernetes_cluster.compute

### ISSUE THREAD

**Issue by mnaghavi** [infracost/infracost#1982]
### Infracost Version
 `v0.10.10`

### Debug Level Output and the TF file
[debug-level-breakdown (1).txt](https://github.com/infracost/infracost/files/9365576/debug-level-breakdown.1.txt)
[azurerm_kubernetes_cluster.compute.tf.txt](https://github.com/infracost/infracost/files/9365597/azurerm_kubernetes_cluster.compute.tf.txt)

### Expected Behavior

Showing the correct estimated cost for `azurerm_kubernetes_cluster.compute`

### Actual Behavior

Showing $0 for `azurerm_kubernetes_cluster.compute` although it calculates the number of hours correctly for 3 VMs

```
 Name                                                                     Monthly Qty  Unit       Monthly Cost (CAD)

 azurerm_kubernetes_cluster.compute
 └─ default_node_pool
    ├─ Instance usage (pay as you go, Standard_standard_d4s_v4)                 2,190  hours         $0.00    # < --------
    └─ os_disk
       └─ Storage (P6)                                                              3  months        $39.32
```

### Steps to Reproduce

the command that I've used:

```
infracost --log-level debug breakdown --terraform-var-file=config/dev/dev.tfvars --path=.
```

### Additional Context

In this environment, we have 3 VMs and the calculation of hours is correct (`2,190 hours`), but the price shows $0.
Also, there is an interesting warning shown in the debug output:
```
level=warning file="github.com/infracost/infracost/internal/prices/prices.go:103" func=github.com/infracost/infracost/internal/prices.setCostComponentPrice msg="No products found for default_node_pool Instance usage (pay as you go, Standard_standard_d4s_v4), using 0.00"
```

**Comment by vdmgolub** [comment]
@mnaghavi Hello! Thank you for providing the debug output and the TF example, it helped a lot to debug the issue. According to the warning message your pool's `vm_size` is `standard_d4s_v4`, however Infracost CLI expects the value to be `Standard_D4s_v4` (case-sensitive). Because it doesn't match the type properly, it makes a price search for non-existing `Standard_standard_d4s_v4` instance type and obviously receives no results.

The solution for this will be modifying your `var.compute_size` variable to be `Standard_D4s_v4`.

I guess we could update the logic to perform case-insensitive matching [here](https://github.com/infracost/infracost/blob/master/internal/providers/terraform/azure/linux_virtual_machine.go#L60), but immediate fix would be updating your variable.

Please let us know if this helps.

**Comment by mnaghavi** [comment]
Hey @vdmgolub, 
Thanks, that's a good work around and now it calculates the costs.

```
# infracost --log-level info breakdown --terraform-var-file=config/dev/dev.tfvars --path=.                                                                                                     
time="2022-08-19T14:43:44-04:00" level=info msg="Evaluating Terraform directory at ."
time="2022-08-19T14:43:44-04:00" level=info msg="Starting: Downloading Terraform modules"
  Warning: Input values were not provided for following Terraform variables: 'variable.client_secret'. Use --terraform-var-file or --terraform-var to specify them.
time="2022-08-19T14:43:44-04:00" level=info msg="Starting: Evaluating Terraform directory"
time="2022-08-19T14:43:44-04:00" level=info msg="Starting: Retrieving cloud prices to calculate costs"

Project: PilotDataPlatform/azure-infra/infra

 Name                                                            Monthly Qty  Unit      Monthly Cost (CAD)

 azurerm_kubernetes_cluster.compute
 └─ default_node_pool
    ├─ Instance usage (pay as you go, Standard_D4s_v4)                 2,190  hours          $607.27
    └─ os_disk
       └─ Storage (P6)                                                     3  months         $39.68
```

But now the real problem is with this change, terraform wants to destroy the resource and creating it again, which means that it's going to destroy the whole AKS nodes and re-create it again.

```
# terraform plan -var-file=config/dev/dev.tfvars
...
~ default_node_pool {
          - availability_zones           = [] -> null
          - enable_host_encryption       = false -> null
          - enable_node_public_ip        = false -> null
          - fips_enabled                 = false -> null
          ~ kubelet_disk_type            = "OS" -> (known after apply)
          - max_count                    = 0 -> null
          - min_count                    = 0 -> null
            name                         = "agentpool"
          ~ node_labels                  = {} -> (known after apply)
          - node_taints                  = [] -> null
          - only_critical_addons_enabled = false -> null
          ~ orchestrator_version         = "1.22.6" -> (known after apply)
          ~ os_sku                       = "Ubuntu" -> (known after apply)
          - tags                         = {} -> null
          ~ vm_size                      = "standard_d4s_v4" -> "Standard_D4s_v4" # forces replacement
            # (8 unchanged attributes hidden)

...
Plan: 1 to add, 0 to change, 1 to destroy.
        }
```

I think that would be better for me to wait for the case-insensitive matching to be there and try after that.

**Comment by vdmgolub** [comment]
Oh, sorry, I haven't thought about that 😅.

I think there is an easy solution for that even with current v0.10.10: CLI can accept multiple TF var flags. In your case you can feed the `config/dev/dev.tfvars` as is and after that (the order matters) you can pass another flag  with a file (or a separate var) to override the original dev.tfvars. Here's an example:

```tf
# example.tf

provider "azurerm" {
  features {}
  skip_provider_registration = true
}

variable "vm_size" {
  description = "Example variable"
  type        = string
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_kubernetes_cluster" "example" {
  name                = "example1"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  dns_prefix          = "example1"

  default_node_pool {
    name    = "default"
    vm_size = var.vm_size
  }
}
```

```
# example.tfvars

vm_size="standard_d4s_v4"
```

```bash
infracost breakdown --path example/terraform --terraform-var-file=example.tfvars

Project: infracost/infracost/examples/terraform

 Name                                                            Monthly Qty  Unit    Monthly Cost

 azurerm_kubernetes_cluster.example
 └─ default_node_pool
    ├─ Instance usage (pay as you go, Standard_standard_d4s_v4)          730  hours          $0.00
    └─ os_disk
       └─ Storage (P10)                                                    1  months        $21.68

 OVERALL TOTAL                                                                              $21.68
```

Now I'll pass an override:
```bash
infracost breakdown --path example/terraform --terraform-var-file=example.tfvars --terraform-var="vm_size=Standard_D4s_v4"

Project: infracost/infracost/examples/terraform

 Name                                                   Monthly Qty  Unit    Monthly Cost

 azurerm_kubernetes_cluster.example
 └─ default_node_pool
    ├─ Instance usage (pay as you go, Standard_D4s_v4)          730  hours        $167.90
    └─ os_disk
       └─ Storage (P10)                                           1  months        $21.68

 OVERALL TOTAL                                                                    $189.58
```

This override doesn't affect your deployment anyhow, and is used only by the CLI.

It might take some time to release a new version, so until we have a code fix, I hope this can unblock you without the need to change anything in your infrastructure :)

**Comment by mnaghavi** [comment]
Thanks @vdmgolub, it's a nice temporary fix and would work properly.

**Comment by vdmgolub** [comment]
@mnaghavi Thank you for fixing this! It should go live with the next release. 👍

**Comment by hugorut** [comment]
Hey @mnaghavi the fix for this has been published in the [0.10.11](https://github.com/infracost/infracost/releases/tag/v0.10.11) release. Thanks for bringing it to our attention.
